cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactTable.java

```diff
@@ -121,11 +125,79 @@ public FactScanner scan(FactScan scan) throws Exception {
                              scan.getStartTs(), scan.getEndTs());
   }
 
+  /**
+   * Given a list of previous tags, we return a list of available tags at the child context in the provided time range.
+   * @param previousTags
+   * @param startTs
+   * @param endTs
+   * @return
+   * @throws Exception
+   */
+  public SortedSet<TagValue> getNextTags(List<TagValue> previousTags, long startTs, long endTs) throws Exception {
```

Comment:
return Collection, would be cleaner API. In javadoc, you can say that collection is sorted